src/saltext/vmware/modules/vmc_sddc_host.py

```diff
@@ -0,0 +1,201 @@
+
+    cluster_id: String
+        (Optional) An optional cluster id if the esxs operation has to be on a specific cluster.
+
+    esxs: Array Of String As UUID
+        (Optional) An optional list of ESX IDs to remove.
+
+    strict_placement: Boolean
+        (Optional) An option to indicate if the host needs to be strictly placed in a placement group.
+        Fail the operation otherwise.
+
+    action: String
+        (Optional)
+        If action = 'add', will add the esx.
+
+        If action = 'remove', will delete the esx/esxs bound to a single cluster(Cluster Id is mandatory for non cluster 1 esx remove).
+
+        If action = 'force-remove', will delete the esx even if it can lead to data loss (This is an privileged operation).
+
+        If action = 'addToAll', will add esxs to all clusters in the SDDC (This is an privileged operation).
+
+        If action = 'removeFromAll', will delete the esxs from all clusters in the SDDC (This is an privileged operation).
+
+        If action = 'attach-diskgroup', will attach the provided diskgroups to a given host (privileged).
+
+        If action = 'detach-diskgroup', will detach the diskgroups of a given host (privileged).
+
+        Default behaviour is 'add'
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    For example:
+        .. code::
+
+            {
+                "availability_zone": "us-west-2a",
+                "cluster_id": "e97920ae-1410-4269-9caa-29584eb8cf6d",
+                "esxs": [
+                "94ce40e1-8619-45b5-9817-0f3466d0dc78"
+                ],
+                "num_hosts": 1,
+                "strict_placement": false
+            }
+
+    """
+
+    log.info("Managing host for the SDDC %s", sddc_id)
+    api_base_url = vmc_request.set_base_url(hostname)
+    api_url = "{base_url}vmc/api/orgs/{org_id}/sddcs/{sddc_id}/esxs".format(
+        base_url=api_base_url, org_id=org_id, sddc_id=sddc_id
+    )
+
+    allowed_dict = {
+        "num_hosts": num_hosts,
+        "availability_zone": availability_zone,
+        "cluster_id": cluster_id,
+        "esxs": esxs,
+        "strict_placement": strict_placement,
+    }
+    req_data = vmc_request._filter_kwargs(allowed_kwargs=allowed_dict.keys(), **allowed_dict)
+    params = vmc_request._filter_kwargs(allowed_kwargs=["action"], action=action)
+    request_data = vmc_request.create_payload_for_request(vmc_templates.manage_sddc_host, req_data)
+    return vmc_request.call_api(
+        method=vmc_constants.POST_REQUEST_METHOD,
+        url=api_url,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        description="vmc_sddc_host.manage",
+        data=request_data,
+        params=params,
```

Comment:
**suggestion** these parameters are not being tested (i.e. they can be replaced with `None` and all tests still pass)

Reply:
What I mean is that 

```
params = vmc_request._filter_kwargs(allowed_kwargs=["action"], action=action)
```

Could be replaced with

```
params = None
```

And there are no tests that would detect this problem.

Reply:
Added the required test.